### PR TITLE
fix(plugin-github-deployments): properly parse ghe hostnames

### PR DESCRIPTION
**Pull request by kuangp**
Signed-off-by: Phil Kuang <pkuang@factset.com>

## Hey, I just made a Pull Request!

Properly parse the hostname from URLs. Also support `url` type locations since entities seem to be annotated with this location type by the catalog preprocessor based on my usage with GHE

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by kuangp** [comment]
cc: @anderoo

File: plugins/github-deployments/src/api/index.ts [hunk 1]
```diff
@@ -28,21 +28,22 @@ const getBaseUrl = (
...
 
   const location = parseLocationReference(host);
-  if (location.type !== 'github') {
+  if (location.type !== 'github' && location.type !== 'url') {
     return 'https://api.github.com';
   }
 
-  const config = scmIntegrationsApi.github.byHost(location.target);
+  const hostname = new URL(location.target).hostname;
+  const config = scmIntegrationsApi.github.byHost(hostname);
```
Comment (freben) [inline plugins/github-deployments/src/api/index.ts:36]: Oh but there is a byUrl method, no need for this manual parsing!

Comment (kuangp) [reply plugins/github-deployments/src/api/index.ts]: thanks for the tip!

**Review by freben — APPROVED** [review]
Thanks!

**Pull request merged**